libr/core/cmd_info.c

```diff
@@ -595,10 +595,14 @@ static int cmd_info(void *data, const char *input) {
 				if (z) { playMsg (core, n, z);}\
 				r_core_bin_info (core, x, mode, va, NULL, y);
 		case 'A': // "iA"
-			newline = false;
-			{
-				int mode = (input[1] == 'j')? 'j': 1;
-				r_bin_list_archs (core->bin, mode);
+			if (input[1] == 'j') {
+				r_cons_print ("{");
```

Comment:
uhm, use pj_o inside r_bin_list_archs instead of printing the braces here

Reply:
That will break `rabin2 -Aj`.
https://github.com/radareorg/radare2/blob/38391b38cb11b7b394b3cc4b2abc6c408dab4857/libr/main/rabin2.c#L1111-L1124
See line 1112. Here the call to r_bin_list_archs is wrapped in manually printed "{" and "}". Afterwards there are potentially other keys in this json object.
For example `rabin2 -Aij bins/mach0/fatmach0-3true` prints
```
{"bins":[{"arch":"x86","bits":64,"offset":4096,"size":13792,"machine":"x86 64 all"},{"arch":"x86","bits":32,"offset":20480,"size":13760,"machine":"386"},{"arch":"ppc","bits":32,"offset":36864,"size":13616,"machine":"7400"}],"imports":[{"ordinal":0,"bind":"NONE","type":"FUNC","name":"exit","plt":4294971200},{"ordinal":1,"bind":"NONE","type":"FUNC","name":"dyld_stub_binder","plt":0}]}
```
So your suggestion can only be applied if we pass pj as argument to `r_bin_list_archs`. And also more refactor are needed in rabin2.c.

Reply:
Ok dont overcomplicate this PR for now :) thanks . The proper fix will require more refactoring as you say